corpdir-srv-sync/src/main/java/com/github/bordertech/corpdir/sync/common/ApiKeyIdSynchronisation.java

```diff
@@ -1,19 +1,27 @@
 package com.github.bordertech.corpdir.sync.common;
 
+import com.github.bordertech.corpdir.api.common.ApiKeyIdObject;
 import com.github.bordertech.corpdir.api.response.DataResponse;
+import com.github.bordertech.corpdir.api.service.BasicKeyIdReadOnlyService;
+import com.github.bordertech.corpdir.api.service.BasicKeyIdService;
+import com.github.bordertech.corpdir.sync.api.mapper.ApiModelMapper;
 import java.util.List;
 
 /**
  * Synchronisation service for keyed API object.
  * 
- * @param <S> source service to load data from
- * @param <D> destination service to write data to
+ * @param <S> Read-only Service Api to fetch data
+ * @param <D> Read-Write Service Api to save data
  * @param <A> the keyed API object
+ * @param <M> mapper to copy API data
  * 
  * @author aswinkandula
  * @since 1.0.0
  */
-public interface ApiKeyIdSynchronisation<S, D, A> {

```

Comment:
Would this work? It might make the interface easier to work with...
```
public interface ApiKeyIdSynchronisation<A extends ApiKeyIdObject> {
	BasicKeyIdReadOnlyService<A> getSourceService();
	BasicKeyIdService<A> getDestinationService();
	ApiModelMapper<A> getApiMapper();
	void syncBaseData();
	DataResponse<List<A>> getSourceData();
	void createOrUpdateData(A fromApiData);
}
```

and then it could be extended with

```
public interface ApiVersionKeyIdSynchronisation<A extends ApiVersionable>
		extends ApiKeyIdSynchronisation<A> {
	@Override
	BasicVersionKeyIdReadOnlyService<A> getSourceService();
	@Override
	BasicVersionKeyIdService<A> getDestinationService();
	void syncBaseData(final Long version);
	void syncLinkedData(final Long version);
	void createOrUpdateData(Long version, A fromApiData);
}
```